app/controllers/requests_controller.rb

```diff
@@ -0,0 +1,52 @@
+class RequestsController < ApplicationController
+  before_action :require_login
+
+  def index
+    @requests = Request.all
+  end
+
+  def new
+    @request = Request.new
+  end
+
+  def create
+    @request = Request.new(request_attributes)
+    @request.requester = current_user
+    if @request.save
+      redirect_to request(@request)
+    else
+      flash.new[:alert] = @request.errors.full_messages.join(", ")
+      render 'new'
+    end
+  end
+
+  def show
+    @request = Request.find_by(params[:id])
+  end
+
+  def update
+    request = Request.find_by(params[:id])
+    request.responder = current_user if request.requester != current_user
+    redirect_to request(@request)
+  end
+
+  def destroy
+    request = Request.find_by(params[:id])
+    request.destroy if request.requester == current user
```

Comment:
I think we need a bit more logic in here. We should include a flash message if the request was successfull. We should redirect to somewhere if the request is successful. 

We should include a flash.now if the request is not successful, and we should render that same request.